auth/user_mgt.go

```diff
@@ -561,6 +582,261 @@ func (c *baseClient) getUser(ctx context.Context, query *userQuery) (*UserRecord
 	return parsed.Users[0].makeUserRecord()
 }
 
+// A UserIdentifier identifies a user to be looked up.
+type UserIdentifier interface {
+	String() string
+
+	validate() error
+	matchesUserRecord(ur *UserRecord) (bool, error)
+	populateRequest(req *getAccountInfoRequest) error
+}
+
+// A UIDIdentifier is used for looking up an account by uid.
+//
+// See GetUsers function.
+type UIDIdentifier struct {
+	UID string
+}
+
+func (id UIDIdentifier) String() string {
+	return fmt.Sprintf("UIDIdentifier{%s}", id.UID)
+}
+
+func (id UIDIdentifier) validate() error {
+	return validateUID(id.UID)
+}
+
+func (id UIDIdentifier) matchesUserRecord(ur *UserRecord) (bool, error) {
+	err := id.validate()
+	if err != nil {
+		return false, err
+	}
+	if id.UID == ur.UID {
+		return true, nil
+	}
+	return false, nil
+}
+
+func (id UIDIdentifier) populateRequest(req *getAccountInfoRequest) error {
+	err := id.validate()
```

Comment:
I think it's better to have a single `validate()` method on `getAccountIntoRequest`. Once it has been fully populated, you can call `req.validate()` in the `GetUsers()` function.